nautobot/core/ui/object_detail.py

```diff
@@ -729,6 +733,55 @@ def render_body_content(self, context):
         return result
 
 
+class _BaseTextPanel(Panel):
+    """A panel that renders simple text or markdown"""
+
+    def __init__(
+        self, *, enable_markdown=True, body_content_template_path="components/panel/body_content_text.html", **kwargs
+    ):
+        self.enable_markdown = enable_markdown
+        self.body_content_template_path = body_content_template_path
+        super().__init__(**kwargs)
+
+    def render_body_content(self, context):
+        text_content = self.get_text(context)
+        if self.body_content_template_path:
+            return get_template(self.body_content_template_path).render(
+                {**context, "enable_markdown": self.enable_markdown, "text_content": text_content}
+            )
+        return text_content
+
+    def get_text(self, context):
+        raise NotImplementedError
+
+
+class ObjectTextPanel(_BaseTextPanel):
+    """
+    Panel that renders text or markdown from given object in the context and given field.
+    """
+
+    def __init__(self, *, object_field=None, **kwargs):
+        self.object_field = object_field
+
+        super().__init__(**kwargs)
+
+    def get_text(self, context):
+        if not context.get("object"):
```

Comment:
Let's confirm whether `obj` or `object` are the correct contexts, or whether we need the logic to handle both (https://github.com/nautobot/nautobot/pull/6345#discussion_r1794147571)

Reply:
At least in Circuits I saw `object` in the context. So maybe I should just add backup for `obj` as well.